pkg/api/domains/domains_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package domains
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	v8weblogic "github.com/verrazzano/verrazzano-crd-generator/pkg/apis/weblogic/v8"
+	"github.com/verrazzano/verrazzano-operator/pkg/controller"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutilcontroller"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestInit ensures that a set of domains can be obtained from the provided set of listers.
+func TestInit(t *testing.T) {
+	// GIVEN an array of clusters and a model binding pair
+	Domains = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": getModelBindingPair(),
+	}
+	// create the tests
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testInit",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	// run the tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// WHEN a WLS domain is appended to the cluster and the Init() method is invoked
+			createAndInsertWLSDomain(tt)
+			// THEN the single domain with the expected name will be found in the Domains slice
+			assert.Equal(t, tt.listers, listerSet, "Wrong number of listers")
```

Comment:
FYI you can do this so you don't have to pass t as first param
	assert := assert.New(t)
        assert.Equal(tt.listers, ..)

Reply:
yes - I know.  However, I don't mind too much having the extra argument and was trying, again, to be consistent at least with the context of the "api" sub-packages and this seems to be the more common pattern